src/api/app/models/comment.rb

```diff
@@ -50,18 +50,17 @@ def create_notification(params = {})
 
   # build an array of users, commenting or being mentioned on the commentable of this comment
   def involved_users
-    users = Set.new
-    users_mentioned = Set.new
+    users = []
+    users_mentioned = []
     Comment.where(commentable: commentable).find_each do |comment|
       # take the one making the comment
-      users << comment.user_id
+      users << comment.user.login
       # check if users are mentioned (regexp borrowed from user model - with whitespace removed)
-      comment.body.scan(%r{@([\w\^\-\.#\*\+&'"]*)}).each do |user|
-        users_mentioned << user[0]
+      comment.body.scan(%r{@([\w\^\-\.#\*\+&'"]*)}).each do |user_login|
+        users_mentioned << user_login.first
       end
     end
-    users += User.where(login: users_mentioned.to_a).pluck(:id)
-    users.to_a
+    users |= User.where(login: users_mentioned).pluck(:login)
```

Comment:
[nitpick] I see some refactoring here to improve SQL queries! :nerd_face: (I can send a PR)